src/vdbfusion_ros/VDBVolume_ros.cpp

```diff
@@ -11,30 +11,42 @@
 #include <tf2_sensor_msgs/tf2_sensor_msgs.h>
 
 #include <Eigen/Core>
+#include <happly.h>
 #include <iostream>
 #include <vector>
+#include <array>
 
-#include "igl/write_triangle_mesh.h"
 #include "openvdb/openvdb.h"
 
 namespace {
-std::vector<Eigen::Vector3d> pcl2SensorMsgToEigen(const sensor_msgs::PointCloud2& pcl2) {
-    std::vector<Eigen::Vector3d> points;
-    points.reserve(pcl2.width);
-    sensor_msgs::PointCloud pcl;
-    sensor_msgs::convertPointCloud2ToPointCloud(pcl2, pcl);
+
+template <typename PCLPoint>
+inline bool isPointFinite(const PCLPoint& point) {
+  return std::isfinite(point.x) && std::isfinite(point.y) &&
+         std::isfinite(point.z);
+}
+
+void pclRgbToVec3i(pcl::PointCloud<pcl::PointXYZRGB>& pcl, std::vector<openvdb::Vec3i>& colors,
```

Comment:
The input pointcloud `pcl::PointCloud<pcl::PointXYZRGB>& pcl` can be made a `const` reference instead as it is not supposed to be modified inside the function anyway

Reply:
Also, it would be better if you could explicitly make the function return the colors and points instead of the `void` return type. This fits better the style of code used in this as well as the VDBFusion repo.

Reply:
I modified it to use a tuple but I'm not so sure we're shooting in the foot...
I agree style-wise it does look better, but as it's handling quite a bit of data, ideally we don't want any copy constructor to be called. 
I'm not so sure C++ compiler will  be able to:
 - perfect-forward upon creation of the tuple. Tha implies both ```openvdb::Vec3i``` and ```Eigen::Vector3d``` have move constructors.
 - copy ellision on return (most likely will)

I didn't have time to analyse the assembly that generates nor test memory usage/performance. You might want to take a look at it to make sure no copy constructor is being called.